runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/endpoints/Endpoint.kt

```diff
@@ -17,6 +17,10 @@ import aws.smithy.kotlin.runtime.http.Url
 
  * @property isHostnameImmutable Flag indicating that the hostname can be modified by the SDK client.
  *
+ * @property attributes A grab-bag property map of endpoint attributes. The values here are considered unstable.
```

Comment:
Basically (got clarification from SEP crowd), that they are entirely arbitrary and shouldn't have any expectations placed on them (specific values being set, structure, etc.). In the future. I'll clarify this in the doc.

Reply:
How would a user construct an endpoint (e.g., in their own custom endpoint provider) and fill in these values?

Reply:
Both it and headers are specified within results for rule evaluation.

So, for example, a ruleset might say "if parameter x is an arn, resolve to foo.com. otherwise, resolve to foobar.com, send this header, and throw this value into the property bag". The SDK implementation has to know what values to look for and how to handle them. An example of something in the AWS-specific property bag is information about signing scope.

We'll generate the code for the rulesets, but someone could obviously write their own custom one and programmatically dictate that an endpoint with some arbitrary parameters be returned in whatever case.

Reply:
I was going to comment on the other PR but I was wondering if the addition of a property bag means we can get rid of `AwsEndpoint` and `AwsEndpointResolver`? IIRC the only difference on the endpoint type is the credential scope. It seems worthwhile to see if there is a path to simplify and have `smithy-kotlin` endpoint and endpoint resolver be able to handle AWS use case.